In [1]:
import pandas as pd
df = pd.read_csv("Ecommerce_Data.csv")

df.head()

/usr/local/lib/python3.12/dist-packages/google/colab/_dataframe_summarizer.py:88: UserWarning: Parsing dates in %d/%m/%Y format when dayfirst=False (the default) was specified. Pass `dayfirst=True` or specify a format to silence this warning.
  cast_date_col = pd.to_datetime(column, errors="coerce")


,order_id,customer_id,product_id,category,price,discount,quantity,payment_method,order_date,delivered_date,region,returned,request_date,return_reason,total_amount,shipping_cost,profit_margin,customer_age,customer_gender
0,O100000,C17270,P234890,Home,164.08,0.15,1,Credit Card,23/12/2023,27/12/2023,West,No,NaN,NaN,139.47,7.88,31.17,60,Female
1,O100001,C17603,P228204,Grocery,24.73,0.00,1,Credit Card,3/4/2025,9/4/2025,South,No,NaN,NaN,24.73,4.60,-2.62,37,Male
2,O100002,C10860,P213892,Electronics,175.58,0.05,1,Credit Card,8/10/2024,12/10/2024,North,No,NaN,NaN,166.80,6.58,13.44,34,Male
3,O100003,C15390,P208689,Electronics,63.67,0.00,1,UPI,14/9/2024,20/9/2024,South,No,NaN,NaN,63.67,5.50,2.14,21,Female
4,O100004,C15226,P228063,Home,16.33,0.15,1,COD,21/12/2024,27/12/2024,East,No,NaN,NaN,13.88,2.74,1.15,39,Male


In [2]:
df.shape

(34500, 19)

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34500 entries, 0 to 34499
Data columns (total 19 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   order_id         34500 non-null  object 
 1   customer_id      34500 non-null  object 
 2   product_id       34500 non-null  object 
 3   category         34500 non-null  object 
 4   price            34500 non-null  float64
 5   discount         34500 non-null  float64
 6   quantity         34500 non-null  int64  
 7   payment_method   34500 non-null  object 
 8   order_date       34500 non-null  object 
 9   delivered_date   34500 non-null  object 
 10  region           34500 non-null  object 
 11  returned         34500 non-null  object 
 12  request_date     1903 non-null   object 
 13  return_reason    1903 non-null   object 
 14  total_amount     34500 non-null  float64
 15  shipping_cost    34500 non-null  float64
 16  profit_margin    34500 non-null  float64
 17  customer_age

In [4]:
df.describe()

,price,discount,quantity,total_amount,shipping_cost,profit_margin,customer_age
count,34500.000000,34500.000000,34500.000000,34500.000000,34500.000000,34500.000000,34500.000000
mean,119.391632,0.049291,1.490725,170.008494,6.152120,28.116505,43.474377
std,195.620477,0.069894,0.932270,357.503014,2.389539,53.352947,14.980682
min,1.010000,0.000000,1.000000,0.820000,0.000000,-6.200000,18.000000
25%,16.690000,0.000000,1.000000,19.710000,4.420000,1.500000,31.000000
50%,45.660000,0.000000,1.000000,56.820000,6.090000,10.550000,43.000000
75%,130.950000,0.100000,2.000000,168.530000,7.830000,33.132500,56.000000
max,2930.470000,0.300000,5.000000,12931.800000,15.650000,1536.170000,69.000000


In [5]:
df.columns


Index(['order_id', 'customer_id', 'product_id', 'category', 'price',
       'discount', 'quantity', 'payment_method', 'order_date',
       'delivered_date', 'region', 'returned', 'request_date', 'return_reason',
       'total_amount', 'shipping_cost', 'profit_margin', 'customer_age',
       'customer_gender'],
      dtype='object')

In [6]:
df.isnull().sum()

,0
order_id,0
customer_id,0
product_id,0
category,0
price,0
discount,0
quantity,0
payment_method,0
order_date,0
delivered_date,0


In [7]:
df.duplicated().sum()

np.int64(0)

In [8]:
df.dtypes

,0
order_id,object
customer_id,object
product_id,object
category,object
price,float64
discount,float64
quantity,int64
payment_method,object
order_date,object
delivered_date,object


In [11]:
# Change Datatype of dates from Object

df['order_date'] =pd.to_datetime(df['order_date'], dayfirst=True)
df['delivered_date'] = pd.to_datetime(df['delivered_date'],dayfirst=True)
df['request_date']= pd.to_datetime(df['request_date'], dayfirst=True)

In [12]:
df.dtypes

,0
order_id,object
customer_id,object
product_id,object
category,object
price,float64
discount,float64
quantity,int64
payment_method,object
order_date,datetime64[ns]
delivered_date,datetime64[ns]


In [13]:
df[['order_date','delivered_date','request_date']].head(10)

,order_date,delivered_date,request_date
0,2023-12-23,2023-12-27,NaT
1,2025-04-03,2025-04-09,NaT
2,2024-10-08,2024-10-12,NaT
3,2024-09-14,2024-09-20,NaT
4,2024-12-21,2024-12-27,NaT
5,2024-04-14,2024-04-19,NaT
6,2025-05-20,2025-05-25,NaT
7,2023-09-14,2023-09-18,NaT
8,2023-10-23,2023-10-29,NaT
9,2023-10-31,2023-11-06,NaT


In [17]:
# 1. Count of returned values
df['returned'].value_counts()

,count
returned,
No,32597
Yes,1903


In [20]:
# 2. Missing request dates
df['request_date'].isnull().sum()

np.int64(32597)

In [21]:
# 3. Returned = Yes but request_date is missing
df[(df['returned'] == 'Yes') & (df['request_date'].isnull())]

,order_id,customer_id,product_id,category,price,discount,quantity,payment_method,order_date,delivered_date,region,returned,request_date,return_reason,total_amount,shipping_cost,profit_margin,customer_age,customer_gender


## Missing Value Analysis

The request_date column contains 32,597 missing values.

After validation, these missing values were found only for orders where returned = 'No'.

Since customers who do not return an order do not have a return request date, these missing values are expected and were retained. No imputation was performed because they accurately represent the business process.

In [22]:
# Remove Extra Spaces

text_columns = df.select_dtypes(include=['object']).columns

for col in text_columns:
  df[col] =df[col].str.strip()

In [23]:
# Standardize Text

df['customer_gender'] = df['customer_gender'].str.title()

df['payment_method'] = df['payment_method'].str.title()

df['category'] = df['category'].str.title()

df['region'] = df['region'].str.title()

df['returned'] = df['returned'].str.title()

In [24]:
# Negative Values

df[df['price'] < 0]

df[df['quantity'] <= 0]

df[df['shipping_cost'] < 0]

,order_id,customer_id,product_id,category,price,discount,quantity,payment_method,order_date,delivered_date,region,returned,request_date,return_reason,total_amount,shipping_cost,profit_margin,customer_age,customer_gender


In [25]:
# Date Validation

(df['delivered_date'] < df['order_date']).sum()

np.int64(0)

In [26]:
(df['request_date'] < df['order_date']).sum()

np.int64(0)

In [27]:
# Export Clean Dataset

df.to_csv("Ecommerce_Clean.csv", index=False)

In [28]:
from google.colab import files

files.download("Ecommerce_Clean.csv")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>